utils/build_swift/driver_arguments.py

```diff
@@ -0,0 +1,1295 @@
+        option('--watchos', enable(all_watchos),
+               help='also build for watchOS, but disallow tests that require '
+                    'a watchOS device')
+        option(['--skip-watchos', '--skip-build-watchos'],
+               disable(all_watchos),
+               help='skip everything watchOS-related')
+
+    option('--skip-build-watchos-device',
+           disable('build_watchos_device'),
+           help='skip building Swift stdlibs for watchOS devices (i.e. '
+                'build simulators only)')
+    option('--skip-build-watchos-simulator',
+           disable('build_watchos_simulator'),
+           help='skip building Swift stdlibs for watchOS simulators '
+                '(i.e. build devices only)')
+
+    option('--darwin-deployment-version-osx', set_,
+           default=defaults.DARWIN_DEPLOYMENT_VERSION_OSX,
+           metavar='VERSION',
+           help='minimum deployment target for OS X')
+
+    option('--darwin-deployment-version-ios', set_,
+           default=defaults.DARWIN_DEPLOYMENT_VERSION_IOS,
+           metavar='VERSION',
+           help='minimum deployment target for iOS')
+
+    option('--darwin-deployment-version-tvos', set_,
+           default=defaults.DARWIN_DEPLOYMENT_VERSION_TVOS,
+           metavar='VERSION',
+           help='minimum deployment target for tvOS')
+
+    option('--darwin-deployment-version-watchos', set_,
+           default=defaults.DARWIN_DEPLOYMENT_VERSION_WATCHOS,
+           metavar='VERSION',
+           help='minimum deployment target for watchOS')
+
+    # -------------------------------------------------------------------------
+    in_group('Project build options')
+
+    # Override to force bootstrap Ninja
+    option('--build-ninja', enable, default=False,
+           help='build Ninja')
+
+    all_required_project_builds = [
+        'build_cmark',
+        'build_llvm',
+        'build_swift',
+    ]
+
+    all_optional_project_builds = [
+        'build_foundation',
+        'build_llbuild',
+        'build_lldb',
+        'build_libdispatch',
+        'build_libicu',
+        'build_playgroundlogger',
+        'build_playgroundsupport',
+        'build_swiftpm',
+        'build_xctest',
+    ]
+
+    set_defaults(all_required_project_builds, True)
+    set_defaults(all_optional_project_builds, False)
+
+    all_project_builds = (
+        all_required_project_builds +
+        all_optional_project_builds)
+
+    option(['-S', '--skip-build'], enable(all_project_builds),
+           help='generate the build directory without building')
+
+    with mutually_exclusive_group():
+        option('--build-cmark',
+               enable('build_cmark'),
+               help='build CommonMark')
+        option('--skip-build-cmark',
+               disable('build_cmark'),
+               help='skip building CommonMark')
+
+    with mutually_exclusive_group():
+        option('--build-llvm',
+               enable('build_llvm'),
```

Comment:
Just to double check (I didn't look), did you preserve the behavior where --skip-build-llvm builds enough of LLVM to compile Swift archives? (I think it tablegens some things and produces a few utilities, but thats it).